soc/silabs/silabs_siwx91x/Kconfig

```diff
@@ -62,4 +62,13 @@ config SIWX91X_ENCRYPT
 	  The key used is the value of CONFIG_SIWX91X_MIC_KEY. This value is
 	  passed as-is to the command "commander rps convert".
 
+config WIFI_SILABS_SIWX91X_FIRMWARE_UPGRADE
+	bool "Implementation of firmware upgrade"
+	select WISECONNECT_NETWORK_STACK
+	help
+	  Enable this option to allow to use of the Firmware upgrade Processor using
+	  zephyr native stack APIs provided by the Network Co-Processor present on
+	  SiWx91x series.Use samples/net/sockets/http_client & samples/net/sockets/tcp
+	  for downloading the data.
```

Comment:
Why specifically those two samples and nothing else? They don't seem to be related to firmware upgrade, rather just generic networking samples. Does enabling this new Kconfig option somehow magically make firmware upgrade available? Don't you also need to call the new APIs that are being imported into the HAL tree? Where's the documentation for that? I guess we don't have any code in Zephyr currently calling those APIs, in which case it'd make sense to have some reference to wiseconnect API documentation, right? What could also be helpful, would be to introduce a sample app to `samples/boards/silabs/` to demonstrate this functionality, but not as an alternative to documentation, rather to complement the documentation.